src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,21 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User cannot be null");
+        }
+        if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new RegistrationException("Login must be at least 6 characters");
```

Comment:
The check for the login length should be split into separate conditions. One for checking if the login is null and another one for checking the length. This allows for more specific error messages, which is in line with the checklist's recommendation to avoid one big if with all conditions.